src/mode/modeHandler.ts

```diff
@@ -707,6 +707,10 @@ export class ModeHandler implements vscode.Disposable {
     if (result instanceof Position) {
       vimState.cursorPosition = result;
     } else if (isIMovement(result)) {
+      if (result.failed) {
+        vimState.recordedState = new RecordedState();
```

Comment:
I believe this is the line that is causing #474, since the stacktrace we're getting is coming from `modeHandler.ts`:

```
    if (!recordedState.operator) {
      throw new Error("what in god's name");
    }
```

But more guessing than actually understanding what "recorded state" is.